odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -305,21 +305,49 @@ def test_20_float(self):
         record.number = 2.4999999999999996
         self.assertEqual(record.number, 2.50)
 
-    def test_20_monetary(self):
-        """ test monetary fields """
+    def test_20_monetary_assign(self):
```

Comment:
you should considere more test cases with 2, 3 and 4 decimals, historically it has been a pain on several cases